src/contacts/model/ContactModel.js

```diff
@@ -73,19 +74,25 @@ export class ContactModelImpl implements ContactModel {
 	 */
 	searchForContacts(query: string, field: string, minSuggestionCount: number): Promise<Contact[]> {
 		return this._worker.search(query, createRestriction("contact", null, null, field, null), minSuggestionCount)
-		             .then(result => {
-			             // load one by one because they may be in different lists when we have different lists
-			             return Promise.map(result.results, idTuple => {
-				             return this._entityClient.load(ContactTypeRef, idTuple).catch(NotFoundError, e => {
-					             return null
-				             }).catch(NotAuthorizedError, e => {
-					             return null
-				             })
-			             }).filter(contact => contact != null)
-		             })
+		           .then(result => {
+			           const resultsByListId = groupBy(result.results, listIdPart)
+			           return Promise.map(resultsByListId, ([listId, idTuples]) => {
+				           // we try to load all contacts from the same list in one request
+				           // however one error will cause the list not to be loaded
+				           return this._entityClient.loadMultipleEntities(ContactTypeRef, listId, idTuples.map(elementIdPart))
+				                      .catch(NotFoundError, e => {
```

Comment:
So this gets triggered as soon as a single contact in that list is not found?
Should we try to retrieve one-by-one then or is that an invalid case?

Reply:
`loadMultipleEntities` won't throw a `NotFoundError`, nonexistent entities will actually just not be returned, so we don't actually need this check at all

Reply:
Yes that's true. I removed the check